daemon/create_windows.go

```diff
@@ -2,20 +2,41 @@ package daemon
 
 import (
 	"fmt"
+	"strings"
 
 	containertypes "github.com/docker/docker/api/types/container"
 	"github.com/docker/docker/container"
 	"github.com/docker/docker/pkg/stringid"
 	"github.com/docker/docker/volume"
 )
 
+// dedupConfigEnvVars deduplicates requested environment variables in the config
+// as they are case insensitive on Windows
+func dedupConfigEnvVars(config *containertypes.Config) {
+	deduped := make(map[string]string)
+	for _, v := range config.Env {
+		i := strings.Index(v, "=")
+		if i >= 0 {
+			name := strings.ToUpper(v[:i])
+			value := v[i+1:]
+			deduped[name] = value
+		}
+	}
+	config.Env = nil
+	for k, v := range deduped {
+		config.Env = append(config.Env, k+"="+v)
+	}
+}
```

Comment:
Doing the dedup in docker and never getting dups into the config feels the right thing IMO. Relying on shell behaviour is (potentially) dangerous. This way we are not leaving anything open to interpretation.